thingif/src/main/java/com/kii/thingif/temporal/base/BaseAnd.java

```diff
@@ -0,0 +1,8 @@
+package com.kii.thingif.temporal.base;
+
+import java.util.List;
+
+public interface BaseAnd<T> {
```

Comment:
What is the purpose to use generic for `BaseAnd`.  I think we don't need it. My exception is 

```java
// if interface is needed
public interface BaseAnd {
   List<BaseClause> getClauses();
   void addClause(BaseClause clause);
}
```